modules/library/jdbc/src/main/java/org/geotools/jdbc/MetadataTablePrimaryKeyFinder.java

```diff
@@ -112,17 +112,9 @@ public PrimaryKey getPrimaryKey(JDBCDataStore store, String schema, String table
                 synchronized (this) {
                     if (metadataTableExists == null) {
                         try {
-
-                            StringBuffer sb = new StringBuffer();
-                            sb.append("SELECT * FROM ");
-                            if (metadataSchema != null) {
-                                store.getSQLDialect().encodeSchemaName(metadataSchema, sb);
-                                sb.append(".");
-                            }
-                            sb.append(tableName).append(" WHERE 1 = 0");
-                            st = cx.prepareStatement(sb.toString());
-                            rs = st.executeQuery();
-                            metadataTableExists = true;
+                            //This is a database-agnostic way to check if the table exists without resulting in error messages
+                            ResultSet resultSet = cx.getMetaData().getTables(null, metadataSchema, tableName, new String[] {"TABLE"});
+                            metadataTableExists = resultSet.next();
```

Comment:
I think this will break the case where GT_PK_METADATA is actually a VIEW instead of a table

Reply:
That's an excellent point @mprins, thank you. We will also make sure we had an online test with a view.

Reply:
One minor thing, instead of using ``resultSet.next()``we could use ``resultSet.hasNext()`` this will have two advantages:
- We will need pay the cost of throwing and caching an exception.
- We will be able to differentiate between:
  - The table doesn't exists, I woudl log this at DEBUG level..
  - We got an exception when looking for the table, I would log this exception as an WARNING and mention that we assume it doesn't exists and move forward.
- I woudl make sure to add proper LOGGING differentiating this two solutions.

Reply:
I addressed the reviewers' comments by making sure the VIEW case is covered, and added a test for this case.
Also added some logic to deal with case-sensitivity.

@nmco I may be mistaken, but I don't think the `resultSet.hasNext()` method exists?

Reply:
Confirming there is not ``hasNext``, ``next`` is used to move to the next record (eventually the first, always needs to be called first) and the boolean reterned is used to check if a record is there, or not (not sure why they designed it differently than iterators, but that's how it is...).